src/ucp/wireup/select.c

```diff
@@ -1508,6 +1508,30 @@ ucp_wireup_add_fast_lanes(ucp_worker_h worker,
     return num_lanes;
 }
 
+/* Restrict choosing more lanes that were chosen previously to prevent
+ * endpoint reconfiguration error in case of dropping some lanes due to low BW.
+ */
+static void
+ucp_wireup_bw_lanes_check_prev(const ucp_wireup_select_params_t *select_params,
```

Comment:
`static unsigned ucp_wireup_get_num_lanes(const ucp_ep_config_key_t *ep_cfg_key, ucp_lane_type lane_type)`

Reply:
As I understand, it is not guaranteed that EP would have assigned key in that point, so we cannot just use `ucp_ep_config(ep)` to pass it. Do you suggest to move `(ep->cfg_index != UCP_WORKER_CFG_INDEX_NULL)` outside of this function or maybe pass key to `ucp_wireup_search_lanes` as argument?

Reply:
yes, move it outside the function

Reply:
As it was discussed, changed  the return type to unsigned + passed EP as input parameter.